File: 프로그래머스/0/120883. 로그인 성공？/로그인 성공？.swift
import Foundation

func solution(_ id_pw:[String], _ db:[[String]]) -> String {
    
    var dic:[String: String] = [:]
    
    db.forEach {
        dic.updateValue($0[1], forKey: $0[0])
    }
    
    if !dic.keys.contains(id_pw[0]) {
        return "fail"
    } else {
        return dic[id_pw[0]] == id_pw[1] ? "login" : "wrong pw"
    }
}

File: 프로그래머스/0/120883. 로그인 성공？/로그인 성공？.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> id_pw, vector<vector<string>> db) {
    string answer = "fail";

    for (const vector<string> &v : db) {
        if(v[0] == id_pw[0]) {
            answer = "wrong pw";
            if(v[1] == id_pw[1]) {
                answer = "login";
                break;
            }
        }
    }

    return answer;
}